signer/src/storage/model.rs

```diff
@@ -81,11 +81,12 @@ pub struct DepositRequest {
 
 impl From<Deposit> for DepositRequest {
     fn from(deposit: Deposit) -> Self {
-        let tx_input_iter = deposit.tx.input.into_iter();
+        let tx_input_iter = deposit.tx_info.vin.into_iter();
```

Comment:
Is there a reason to prefer `tx_info.vin` to `tx_info.tx.input`? Note that in the test below, in `add_deposit`, we are not populating `vin`.

Reply:
> Is there a reason to prefer `tx_info.vin` to `tx_info.tx.input`? 

Yes, `tx_info.tx.input` is the "raw" transaction's input, as stored on a bitcoin block. The raw inputs do not include the `scriptPubKey` of the inputs, which is what we want here.

> Note that in the test below, in `add_deposit`, we are not populating `vin`.

Yeah, it wasn't necessary for those tests and it's non-trivial to populate correctly.